examples/react/module-api-restful/src/components/user-form.tsx

```diff
@@ -0,0 +1,53 @@
+import { useState } from "react";
+import {
+  useUser,
+  useAddUser,
+  useRemoveUser,
+  useUpdateUser,
+} from "../hooks/use-user";
+
+type Props = {
+  id?: string;
+};
+
+export function UserForm({ id }: Props) {
+  const isUpdate = !!id;
+
+  const { data, isLoading } = useUser(id);
+  const { mutate: addMutate } = useAddUser();
+  const { mutate: updateMutate } = useUpdateUser();
+  const { mutate: removeMutate } = useRemoveUser();
+
+  const [value, setValue] = useState(data?.name || "");
+
+  const onAddSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    addMutate({ name: value });
+  };
+
+  const onUpdateSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    id && updateMutate({ id, name: value });
+  };
+
+  const onRemoveClick = async () => {
+    id && removeMutate(id);
+  };
+
+  if (id && isLoading) return <span aria-label="loading">Loading...</span>;
+
+  return (
+    <>
+      <form onSubmit={isUpdate ? onUpdateSubmit : onAddSubmit}>
+        <input value={value} onChange={(e) => setValue(e.target.value)} />
```

Comment:
[IMO] It would be a better example to attach a proper name to this input field. i.e. `name` or `userName`.

Perhaps the `useState` on line 21 could also be changed to reflect a more realistic `user` data model?

```ts
const [user, setUser] = useState({ name: data?.name || ""));
```